Elevel/Application/Features/AuditionCommands/UpdateAuditionCommand.cs

```diff
@@ -63,13 +94,29 @@ public Handler(IApplicationDbContext context, IMapper mapper)
 
             public async Task<Response> Handle(Request request, CancellationToken cancellationToken)
             {
+                if (request.AudioFilePath != null && !File.Exists(request.AudioFilePath))
+                {
+                    throw new NotFoundException($"File with path {request.AudioFilePath} not found.");
+                }
+
                 var audition = await _context.Auditions
-                    .ProjectTo<Response>(_mapper.ConfigurationProvider)
+                    .Include(x => x.Questions)
+                    .ThenInclude(x=>x.Answers)
                     .FirstOrDefaultAsync(x => x.Id == request.Id, cancellationToken);
                 
                 if (audition is null)
                 {
-                    throw new NotFoundException($"Audition with id {request.Id}");
+                    throw new NotFoundException($"Audition with id {request.Id} not found");
+                }
+
+                foreach (var question in request.Questions)
+                {
+                    var dbQuestion = audition.Questions.FirstOrDefault(x => x.Id == question.Id);
+                    if (dbQuestion != null)
+                    {
+                        question.QuestionNumber = dbQuestion.QuestionNumber;
+                        question.Level = request.Level;
+                    }
                 }
 
                 audition = _mapper.Map(request, audition);
```

Comment:
You will receive an Exception with cascade delete
Do it, like you do with question
Don't forget about id validation

Reply:
Fixed by Automapper configuration